app/lib/server/functions/addUserToRoom.js

```diff
@@ -21,6 +22,10 @@ export const addUserToRoom = function(rid, user, inviter, silenced) {
 		return;
 	}
 
+	if (room.t === 'l' && !hasPermission(user._id, 'view-l-room')) {
```

Comment:
please move this code to `allowMemberAction` method of the livechat room type.. the idea behind it is specifically to not have this kind specific code spread across common function.

Reply:
Yeah, that was the idea but the signature of the `allowMemberAction` method doesn't support the user who is going to join the room, which can be added, then I can move the code to `allowMemberAction` method and check all conditions there.

Reply:
no, we didn't 😬  we have a `beforeLeaveRoom` https://github.com/RocketChat/Rocket.Chat/blob/b469c18d39bf867af4da1e75e55bd6bee6e15fd5/app/livechat/server/startup.js#L77

Reply:
Oh nice! The callback seems to be a good option here. 
Let me change the code and see how it works! :D

Reply:
@sampaiodiego the fixed based on callback is done!